src/modules/Dashboard/features/RespondentData/RespondentData.types.ts

```diff
@@ -34,8 +34,24 @@ export type NumberSelectionResponseValues = {
   maxValue: number;
 };
 
+export type PerRowSelectionItemRow = {
+  id: string;
+  rowImage: string | null;
+  rowName: string;
+  tooltip: string | null;
+};
+
+export type PerRowSelectionItemOption = {
+  id: string;
+  image: string | null;
+  text: string;
+  tooltip: string | null;
+  value?: string | number | null;
+};
+
 export type ItemResponseValues = {
-  options: ItemOption[];
+  options: ItemOption[] | PerRowSelectionItemOption[];
```

Comment:
Let's introduce the union of types. It will reduce complexity of type detection and determination of which type should be for which item, for ex.:
``` 
export type ItemResponseValues = SimpleItemResponseValues 
 | NumberSelectionResponseValues | SingleAndMultiSelectionPerRowResponseValues
```